### PR TITLE
chore: remoteWrite add more info (address) when showing errors

### PR DESCRIPTION
**Pull request by chengjoey**
#1961  remoteWrite add more info (url) when showing errors

now, the error msg like:
```
ERRO[2023-05-26T17:59:58.267651] /pkg/remotewrite/queue.go:96 error happened while ingesting data           error="response not ok: status code: '403'. body: ''. url: 'http://localhost:5000/ingest'" key="pyroscope.server{}" remote_target=custom-write
```

### CODE REVIEW

**Comment by chengjoey** [comment]
@eh-am

**Comment by eh-am** [comment]
Thank you!

That's pretty helpful, although I think also having the >exact< URL it's trying to hit would help even more.
Reason is that I was debugging something and I was getting 404, and didn't know what the exact URL was being hit. Did it had `/ingest`? Did it have `/pyroscope`? Did it make an error when concatenating remoteAddress with + `/ingest`? Etc.

That being said, can you add the actual URL being used here? https://github.com/grafana/pyroscope/blob/f710852d6c8b7b68c9dc3c06d0afae647dc0aa16/pkg/remotewrite/client.go#L96

Thanks!

**Comment by chengjoey** [comment]
> 

hi @eh-am , I can indeed get the real url, which of the following two logs do you think is better：

remoteAddress+path
```
ERRO[2023-05-26T17:54:08.131906] /pkg/remotewrite/queue.go:96 error happened while ingesting data           error="response not ok: status code: '403'. body: ''.  path: '/ingest'" key="pyroscope.server{}" remote_address="http://localhost:5000" remote_target=custom-write
```

url:
```
ERRO[2023-05-26T17:57:50.018383] /pkg/remotewrite/queue.go:96 error happened while ingesting data           error="response not ok: status code: '403'. body: ''. url: 'http://localhost:5000/ingest'" key="pyroscope.server{}" remote_target=custom-write
```

**Comment by eh-am** [comment]
@chengjoey Second one please! When you are looking at errors like that you want as less cognitive process as possible hahaha.

**Comment by chengjoey** [comment]
@eh-am , I choose the second one, please review this pr again~

File: pkg/remotewrite/client.go [hunk 1]
```diff
@@ -93,7 +95,7 @@ func (r *Client) Ingest(ctx context.Context, in *ingestion.IngestInput) error {
 	if !(res.StatusCode >= 200 && res.StatusCode < 300) {
 		// read all the response body
 		respBody, _ := ioutil.ReadAll(res.Body)
-		return fmt.Errorf("%w: %v", ErrNotOkResponse, fmt.Errorf("status code: '%d'. body: '%s'", res.StatusCode, respBody))
+		return fmt.Errorf("%w: %v", ErrNotOkResponse, fmt.Errorf("status code: '%d'. body: '%s'. url: '%s'", res.StatusCode, respBody, r.url))
```
Comment (eh-am) [inline pkg/remotewrite/client.go:98]: Let's use [`r.URL.Redacted()`](https://pkg.go.dev/net/url#URL.Redacted) to not leak basic auth password.

https://go.dev/play/p/2XzEbHtX-OP

Comment (chengjoey) [reply pkg/remotewrite/client.go]: done, thanks! @eh-am

**Review by eh-am — APPROVED** [review]
Thank you so much! Sorry for the back and forth, I didn't realize the basic auth case until later!

**Pull request merged**